examples/contrib/detection/faster_rcnn.py

```diff
@@ -0,0 +1,286 @@
+    def log_validation_results(engine):
+        evaluator.run(val_dataloader)
+        visualizer.run(vis_dataloader)
+
+    @trainer.on(Events.ITERATION_COMPLETED)
+    def step_scheduler(engine):
+        scheduler.step()
+        aim_logger.log_metrics({"lr": scheduler.get_last_lr()[0]}, step=engine.state.iteration)
+
+    @visualizer.on(Events.EPOCH_STARTED)
+    def reset_vis_images(engine):
+        engine.state.model_outputs = []
+
+    @visualizer.on(Events.ITERATION_COMPLETED)
+    def add_vis_images(engine):
+        engine.state.model_outputs.append(engine.state.output)
+
+    @visualizer.on(Events.ITERATION_COMPLETED)
+    def submit_vis_images(engine):
+        aim_images = []
+        for outputs in engine.state.model_outputs:
+            for image, target, pred in zip(outputs["x"], outputs["y"], outputs["y_pred"]):
+                image = (image * 255).byte()
+                pred_labels = [Dataset.class2name[l.item()] for l in pred["labels"]]
+                pred_boxes = pred["boxes"].long()
+                image = draw_bounding_boxes(image, pred_boxes, pred_labels, colors="red")
+
+                target_labels = [Dataset.class2name[l.item()] for l in target["labels"]]
+                target_boxes = target["boxes"].long()
+                image = draw_bounding_boxes(image, target_boxes, target_labels, colors="green")
+
+                aim_images.append(aim.Image(image.numpy().transpose((1, 2, 0))))
+        aim_logger.experiment.track(aim_images, name="vis", step=trainer.state.epoch)
+
+    losses = ["loss_classifier", "loss_box_reg", "loss_objectness", "loss_rpn_box_reg", "loss_average"]
+    for loss_name in losses:
+        RunningAverage(output_transform=lambda x: x[loss_name]).attach(trainer, loss_name)
+    ProgressBar().attach(trainer, losses)
+    ProgressBar().attach(evaluator)
+
+    objects_to_checkpoint = {"trainer": trainer, "model": model, "optimizer": optimizer, "lr_scheduler": scheduler}
+    checkpoint = Checkpoint(
+        to_save=objects_to_checkpoint,
+        save_handler=DiskSaver(log_dir, require_empty=False),
+        n_saved=3,
+        score_name="mAP",
+        global_step_transform=lambda *_: trainer.state.epoch,
+    )
+    evaluator.add_event_handler(Events.EPOCH_COMPLETED, checkpoint)
+
+    aim_logger.log_params(
+        {
+            "lr": lr,
+            "image_size": image_size,
+            "batch_size": batch_size,
+            "epochs": epochs,
+        }
+    )
+    aim_logger.attach_output_handler(
+        trainer, event_name=Events.ITERATION_COMPLETED, tag="train", output_transform=lambda loss: loss
+    )
+    aim_logger.attach_output_handler(
+        evaluator,
+        event_name=Events.EPOCH_COMPLETED,
+        tag="val",
+        metric_names=["mAP"],
+        global_step_transform=global_step_from_engine(trainer, Events.ITERATION_COMPLETED),
+    )
+
+    trainer.run(train_dataloader, max_epochs=epochs)
+
+
+if __name__ == "__main__":
+    parser = ArgumentParser()
```

Comment:
Just a question if it won't be more easy to use [python-fire](https://github.com/google/python-fire) instead of arg parser ?